### PR TITLE
Fix cpplint warnings

### PR DESCRIPTION
**Pull request by esteve**
Fix cpplint warnings.

Connects to https://github.com/ros2/ros2/issues/12

### CODE REVIEW

File: test_rclcpp/test/test_allocator.cpp [hunk 1]
```diff
@@ -59,9 +59,9 @@ static inline bool check_stacktrace(const char ** tokens, size_t num_tokens, siz
 
   // allocate string which will be filled with the demangled function name
   size_t funcnamesize = 256;
-  char * funcname = (char *)malloc(funcnamesize);
+  char * funcname = reinterpret_cast<char *>(malloc(funcnamesize));
```
Comment (dirk-thomas) [inline test_rclcpp/test/test_allocator.cpp]: A `void *` should be statically casted - not reinterpreted.

File: test_rclcpp/test/test_allocator.cpp [hunk 2]
```diff
@@ -253,13 +253,13 @@ struct InstrumentedAllocator : public allocator_pointer_traits<T>
...
-  // construct and destroy should not be required to implement allocator_traits, this is a bug in gcc 4.8
-  // for usage in std::pair, std::map
+  // construct and destroy should not be required to implement allocator_traits, this is a bug in
+  // gcc 4.8 for usage in std::pair, std::map
   template<typename _Up, typename ... _Args>
   void
   construct(_Up * __p, _Args && ... __args)
   {
-    ::new((void *)__p)_Up(std::forward<_Args>(__args) ...);
+    ::new(reinterpret_cast<void *>(__p))_Up(std::forward<_Args>(__args) ...);
```
Comment (dirk-thomas) [inline test_rclcpp/test/test_allocator.cpp]: `static_cast`?

File: test_rclcpp/test/test_allocator.cpp (hunk 1 above)
Comment (esteve) [inline test_rclcpp/test/test_allocator.cpp]: That depends on the case, but here a `static_cast` is enough. Updated.

File: test_rclcpp/test/test_allocator.cpp [hunk 3]
```diff
@@ -253,13 +253,13 @@ struct InstrumentedAllocator : public allocator_pointer_traits<T>
     std::free(ptr);
   }
 
-  // construct and destroy should not be required to implement allocator_traits, this is a bug in gcc 4.8
-  // for usage in std::pair, std::map
+  // construct and destroy should not be required to implement allocator_traits, this is a bug in
+  // gcc 4.8 for usage in std::pair, std::map
```
Comment (dirk-thomas) [inline test_rclcpp/test/test_allocator.cpp]: Can you wrap this after the comma?

**Comment by dirk-thomas** [comment]
Beside the late comment: +1

File: test_rclcpp/test/test_allocator.cpp (hunk 3 above)
Comment (esteve) [inline test_rclcpp/test/test_allocator.cpp]: Done.

**Pull request merged**